ktor-server/ktor-server-plugins/ktor-server-call-logging/jvm/src/io/ktor/server/plugins/callloging/MDCEntryUtils.kt

```diff
@@ -29,7 +29,8 @@ internal fun List<MDCEntry>.setup(call: ApplicationCall): Map<String, String> {
     val result = HashMap<String, String>()
 
     forEach { entry ->
-        entry.provider(call)?.let { mdcValue ->
+        val provider = runCatching { entry.provider(call) }.getOrNull()
```

Comment:
maybe at least log error here?

Reply:
Nope, we shouldn't.

We introduced this problem by intercepting several phases to put MDC in, and in the earliest phase session access will fail because they were not handled at that moment. It leads to an exception and a whole request failure.